### PR TITLE
[ROCm] Add ROCm support for remaining cwise ops and tests

### PR DESCRIPTION
**Pull request by jeffdaily**
cwise_ops.h -- packet access is always false for ROCm.
Some ROCm ops do not yet support complex64 / complex128 types.

### CODE REVIEW

File: tensorflow/core/kernels/cwise_ops.h [hunk 1]
```diff
@@ -176,7 +176,11 @@ template <typename T>
 struct functor_traits<div_no_nan_op<T>> {
   enum {
     Cost = functor_traits<scalar_quotient_op<T>>::Cost + NumTraits<T>::AddCost,
+#if TENSORFLOW_USE_ROCM
```
Comment (chsigg) [inline tensorflow/core/kernels/cwise_ops.h]: Would it make sense to add 'IsRocm()', 'IsCuda()' functions somewhere?
Separate PR, if feasible at all.

File: tensorflow/core/kernels/cwise_ops.h [hunk 2]
```diff
@@ -350,8 +366,13 @@ struct google_floor_div {
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const T operator()(const T& x,
                                                            const T& y) const {
     if ((x < T(0)) != (y < T(0))) {
+#if defined(__HIP_DEVICE_COMPILE__)
```
Comment (chsigg) [inline tensorflow/core/kernels/cwise_ops.h]: Could you add a comment why we need this special handling?

File: tensorflow/core/kernels/cwise_ops.h (hunk 1 above)
Comment (jeffdaily) [reply tensorflow/core/kernels/cwise_ops.h]: Are you wanting sometthing like 'IsRocm()' to resolve at run-time or compile-time?

Comment (chsigg) [reply tensorflow/core/kernels/cwise_ops.h]: Something like: 
```bool isRocm() {
#if TENSORFLOW_USE_ROCM
  return true;
#endif
  return false;
}```

**Review by chsigg — APPROVED** [review]

**Pull request merged**